### PR TITLE
Feature/error handling

### LINKED ISSUE #29: Error Handling
- [x] Implement Error Handling for the main page, single page, and submission
- [ ] On the home page, the home button should not show 
- [x] When the user searches for something that's not there, show GIF
- [x] Server error messaging and GIF (POST, any GETS)
- [x] Add required tags to inputs for posting a review 
- [x] Disable submit review button until the user has filled in the inputs

### PR DESCRIPTION
**Pull request by stephanie-roe**
# Description
Please include a summary of the changes and/or which issue is fixed. Please also include relevant motivation and context. 
- Added error handling for search bar, review form and fetches. 

Were there any issues that arose? 
- n/a 

Is there anything you need from your teammates? 
- n/a

Any other comments, questions, or concerns?
- n/a 

## Issues
Closes #29 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactor feature (non-breaking change that modifies existing work)
- [ ] Cosmetic changes (changes to layout or visuals in CSS file)

# How Has This Been Tested?
- [x] React Dev Tools
- [x] Functionality check through web browser
- [ ] Ran Cypress test suite

# Checklist:
- [x] My code follows the style guidelines of this project (semicolons, indentation, naming conventions, etc..).
- [x] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have deleted all comments after functionality and understanding is final.
- [x] I have added any issues closed by this PR in the Issues section.
- [x] I have added others as reviewers.

### CODE REVIEW

File: src/Components/App.tsx [hunk 1]
```diff
@@ -136,27 +133,31 @@ class App extends React.Component<{}, State> {
...
+          <NavBar searchBrewery={this.searchBrewery} clearSearchBreweries={this.clearSearchBreweries} query={this.state.query}/>
+          <Switch>
+          <Route exact path="/"
+            render={() => {
+              if (!this.state.searchedBreweries.length && !this.state.query) {
+                return (<Breweries filterReviews={this.filterBreweryReviews} newBrewery={this.state.breweries} />)
+              } else if (!this.state.searchedBreweries.length) {
+                return (<ErrorMessage/>)
+              } else {
+                return (<Breweries filterReviews={this.filterBreweryReviews} newBrewery={this.state.searchedBreweries} />)
```
Comment (rachellhill) [inline src/Components/App.tsx:150]: Great work error handling!

File: src/Components/ErrorMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+
+const ErrorMessage = (): JSX.Element => {
+    return (
+        <div>
+            <img src="https://media.giphy.com/media/l46Cl6JOKu0fbiR3O/giphy.gif" alt="sock puppet drinking beer"></img>
+            <h1>Something went wrong, please try again!</h1>
+        </div>
```
Comment (rachellhill) [inline src/Components/ErrorMessage.tsx:8]: Love the giph!

**Review by aominhlong — APPROVED** [review]

**Review by rachellhill — APPROVED** [review]
Great work error handling!

File: src/Components/ErrorMessage.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
```
Comment (userigorgithub) [inline src/Components/ErrorMessage.tsx:1]: Good job implementing new component ErrorMessage!

**Comment by aominhlong** [comment]
This looks good! Great job on error handling!!

**Review by userigorgithub — APPROVED** [review]
Great job implementing error handling for different situations!

**Pull request merged**